src/jupyterIntegration.ts

```diff
@@ -223,21 +225,53 @@ export class JupyterServerIntegration implements JupyterServerProvider, JupyterS
             )
             .catch(noop);
 
-        const baseUrl = Uri.parse(
-            await getUserJupyterUrl(
-                serverInfo.baseUrl,
-                authInfo.username || '',
-                authInfo.token,
-                this.fetch,
-                cancelToken
-            )
+        const rawBaseUrl = await getUserJupyterUrl(
+            serverInfo.baseUrl,
+            authInfo.username || '',
+            authInfo.token,
+            this.fetch,
+            cancelToken
         );
+        // https://github.com/microsoft/vscode-jupyter-hub/issues/53
+        // baseUrl = baseUrl.replace(/%40/g, '@').replace(/%7E/g, '~');
+        const baseUrl = Uri.parse(rawBaseUrl);
```

Comment:
A set of unit tests might help explain what is expected to happen here. This is pretty difficult to know what's going on here.

Reply:
Agreed